Wrapping/Generators/Python/itkTemplate.py

```diff
@@ -543,14 +543,82 @@ def firstIfList(arg):
         if increase_dimension and imageIO.GetDimensions(dimension-1) != 1:
             dimension += 1
         componentAsString = imageIO.GetComponentTypeAsString(imageIO.GetComponentType())
-        component = componentTypeDic[componentAsString]
+        _io_component_type_dict = {
+                "float": itk.F,
+                "double": itk.D,
+                "unsigned_char": itk.UC,
+                "unsigned_short": itk.US,
+                "unsigned_int": itk.UI,
+                "unsigned_long": itk.UL,
+                "unsigned_long_long": itk.ULL,
+                "char": itk.SC,
+                "short": itk.SS,
+                "int": itk.SI,
+                "long": itk.SL,
+                "long_long": itk.SLL
+                }
+        component = _io_component_type_dict[componentAsString]
         pixel = imageIO.GetPixelTypeAsString(imageIO.GetPixelType())
         numberOfComponents = imageIO.GetNumberOfComponents()
         PixelType = itkTemplate._pixelTypeFromIO(pixel, component, numberOfComponents)
         ImageType = itk.Image[PixelType, dimension]
         ReaderType = TemplateReaderType[ImageType]
         return ReaderType.New(*args, **kwargs)
 
+    @staticmethod
+    def _NewMeshReader(TemplateReaderType, *args, **kwargs):
+        def firstIfList(arg):
+            if type(arg) in [list, tuple]:
+                return arg[0]
+            else:
+                return arg
+        inputFileName = ''
+        if len(args) != 0:
+            # try to find a type suitable for the primary input provided
+            inputFileName = firstIfList(args[0])
+        elif 'FileName' in kwargs:
+                inputFileName = firstIfList(kwargs['FileName'])
+        if not inputFileName:
+            raise RuntimeError("No FileName specified.")
+        import itk
+        if "MeshIO" in kwargs:
+            meshIO = kwargs["MeshIO"]
+        else:
+            meshIO = itk.MeshIOFactory.CreateMeshIO(inputFileName, itk.MeshIOFactory.ReadMode)
+        if not meshIO:
+            msg = ""
+            if not os.path.isfile(inputFileName):
+                msg += ("\nThe file doesn't exist. \n" +
+                       "Filename = %s" % inputFileName)
+            raise RuntimeError("Could not create IO object for reading file %s" % inputFileName + msg)
+        # Read the metadata from the mesh file.
+        meshIO.SetFileName(inputFileName)
+        meshIO.ReadMeshInformation()
+        dimension = meshIO.GetPointDimension()
+        componentAsString = meshIO.GetComponentTypeAsString(meshIO.GetPointPixelComponentType())
+        _io_component_type_dict = {
```

Comment:
This is a duplicated definition, and therefore prone to discrepancies. Can this be shared with `_NewImageReader`'s definition?

Reply:
There is a subtle difference between the definitions -- to keep it cleaner they were defined this way. I added a comment to explain that it is intentional and what was done.